src/CodeGen_PTX_Dev.cpp

```diff
@@ -262,9 +262,15 @@ void CodeGen_PTX_Dev::visit(const Call *op) {
         auto fence_type_ptr = as_const_int(op->args[0]);
         internal_assert(fence_type_ptr) << "gpu_thread_barrier() parameter is not a constant integer.\n";
 
-        llvm::Function *barrier0 = module->getFunction("llvm.nvvm.barrier0");
-        internal_assert(barrier0) << "Could not find PTX barrier intrinsic (llvm.nvvm.barrier0)\n";
-        builder->CreateCall(barrier0);
+        llvm::Function *barrier = module->getFunction("llvm.nvvm.barrier0");
```

Comment:
Suggestion: add a comment that specifically says "this can be removed once we no longer support LLVM19 or earlier"

Reply:
Did that, and reordered the flow of the checks to favor the up-to-date check first.